### PR TITLE
[android] Add warning message when Android devices have package verification enabled

### PR DESCRIPTION
**Pull request by ivanpovazan**
## Description

This PR adds functionality to check if a physical Android device is configured to correctly to allow debug APK installs.
Additionally, it prints out warning messages in cases the package verifier is not properly disabled which can lead to `INSTALL_FAILED_VERIFICATION_FAILURE` errors.

## Follow-up work

- [ ] https://github.com/dotnet/dnceng/issues/4253
- If tests start failing again the warning message can be tracked with Build Analysis to record which Helix/tethered devices are affected

### CODE REVIEW

**Comment by ivanpovazan** [comment]
@akoeplinger could you please take a look at this? 
It is still WIP, but showcases the general idea.
Log of reference: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-xharness-refs-pull-1291-merge-b23bc16565734eeea8/System.Buffers.Tests-arm64-v8a/1/console.4c8eb5eb.log?helixlogtype=result

File: src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs [hunk 1]
```diff
@@ -188,6 +188,62 @@ public int GetDeviceApiVersion()
...
+        var packageVerifierSetting = ReadPackageVerificationSettingValue("package_verifier_enable");
+        if (packageVerifierSetting != 0)
+        {
+            _log.LogInformation($"package_verifier_enable needs to be set to '0' so that debug APKs can be installed");
+            RunAdbCommand(new[] { "shell", "settings", "put", "global", "package_verifier_enable", "0"})
+                .ThrowIfFailed("Error setting package_verifier_enable to '0'");
+            packageVerifierSetting = ReadPackageVerificationSettingValue("package_verifier_enable");
+            if (packageVerifierSetting != 0)
+            {
+                _log.LogError($"Failed to set packageVerifierSetting to '0'");
```
Comment (ivanpovazan) [inline src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs:239]:
```suggestion
                _log.LogError($"Failed to set package_verifier_enable to '0'");

```

**Comment by ivanpovazan** [comment]
We now generate the following log and warning messages in cases package verifier is not properly disabled:
```
[04:11:34] dbug: Check current adb install and/or package verification settings
[04:11:34] dbug: Executing command: 'D:\h\w\9C86086D\p\microsoft.dotnet.xharness.cli\10.0.0-ci\runtimes\any\native\adb\windows\adb.exe -s 26091FDF60024Q shell settings get global verifier_verify_adb_installs'
[04:11:34] dbug: verifier_verify_adb_installs = 
[04:11:34] warn: Installing debug apks on a device might be rejected with INSTALL_FAILED_VERIFICATION_FAILURE. Make sure to set 'verifier_verify_adb_installs' to '0'
[04:11:34] dbug: Executing command: 'D:\h\w\9C86086D\p\microsoft.dotnet.xharness.cli\10.0.0-ci\runtimes\any\native\adb\windows\adb.exe -s 26091FDF60024Q shell settings get global package_verifier_enable'
[04:11:34] dbug: package_verifier_enable = 
[04:11:34] warn: Installing debug apks on a device might be rejected with INSTALL_FAILED_VERIFICATION_FAILURE. Make sure to set 'package_verifier_enable' to '0'
```
Relevant log: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-xharness-refs-pull-1291-merge-cc92462c3409498299/System.Buffers.Tests-arm64-v8a.Attempt.3/1/console.9f873a78.log?helixlogtype=result

**Review by matouskozak — APPROVED** [review]
Neat work!

**Pull request merged**